quisp/modules/QRSA/ConnectionManager/ConnectionManager.cc

```diff
@@ -303,26 +306,42 @@ void ConnectionManager::respondToRequest(ConnectionSetupRequest *req) {
   // create RuleSet for all nodes!
   int num_resource = req->getNumber_of_required_Bellpairs();
   int intermediate_node_size = req->getStack_of_QNodeIndexesArraySize();
+  // generate the rulesets for intermediate swappers
   for (int i = 0; i <= intermediate_node_size; i++) {
     auto itr = std::find(swappers.begin(), swappers.end(), path.at(i));
     size_t index = std::distance(swappers.begin(), itr);
+
     if (index != swappers.size()) {
       EV_DEBUG << "Im swapper!" << path.at(i) << "\n";
       // generate Swapping RuleSet
       // here we have to check the order of entanglement swapping
-
       // swapping configurations for path[i]
-      SwappingConfig config = generateSwappingConfig(path.at(i), path, swapping_partners, qnics, num_resource);
-      RuleSet *rule = generateEntanglementSwappingRuleSet(path.at(i), config);
 
-      ConnectionSetupResponse *pkr = new ConnectionSetupResponse("ConnSetupResponse(Swapping)");
-      pkr->setDestAddr(path.at(i));
-      pkr->setSrcAddr(my_address);
-      pkr->setKind(2);
-      pkr->setRuleSet(rule);
-      pkr->setActual_srcAddr(path.at(0));
-      pkr->setActual_destAddr(path.at(path.size() - 1));
-      send(pkr, "RouterPort$o");
+      if (!simultaneous_es_enabled) {
+        SwappingConfig config = generateSwappingConfig(path.at(i), path, swapping_partners, qnics, num_resource);
+        RuleSet *rule = generateEntanglementSwappingRuleSet(path.at(i), config);
+
+        auto *pkr = new ConnectionSetupResponse("ConnSetupResponse(Swapping)");
+        pkr->setDestAddr(path.at(i));
+        pkr->setSrcAddr(my_address);
+        pkr->setKind(2);
+        pkr->setRuleSet(rule);
+        pkr->setActual_srcAddr(path.at(0));
+        pkr->setActual_destAddr(path.at(path.size() - 1));
+        send(pkr, "RouterPort$o");
+      } else {
+        SwappingConfig config = generateSimultaneousSwappingConfig(path.at(i), path, qnics, num_resource);
+        RuleSet *rule = generateSimultaneousEntanglementSwappingRuleSet(path.at(i), config, path);
```

Comment:
For the later implementation, this should be generalized. (Take "simultaneous" flag as the argument of generateEntanglementSwapppingConfig)